### ISSUE TITLE
Telemetry: Credit card autofill category incorrectly reclassified as metrics

### ISSUE THREAD

**Issue by wlach** [mozilla-mobile/fenix#19976]
The credit card autofill telemetry introduced for #18711 was originally given the category `credit_cards` but somehow got reclassified under the `metrics` category in #19733.

This makes it much harder to identify in Looker. Compare:

<img width="289" alt="image" src="https://user-images.githubusercontent.com/20569/121743211-fe182100-cace-11eb-9a47-766f5f12f259.png">

(the old names)

With:

<img width="313" alt="image" src="https://user-images.githubusercontent.com/20569/121743161-e5a80680-cace-11eb-8f31-74ca6eed9cf7.png">

(the new category)

Note that the telemetry is now also duplicated within the Glean Dictionary:

<img width="1406" alt="image" src="https://user-images.githubusercontent.com/20569/121743351-3ae41800-cacf-11eb-9944-531b76737263.png">

https://dictionary.telemetry.mozilla.org/apps/fenix?itemType=metrics&page=1&search=credit_card&showExpired=1

It is not necessary to give a metric the category `metrics` to make it appear in the metrics ping. 

We will shortly be hiding metrics that are no longer in the source by default in Looker, as well as making it more clear they're no longer available in the Glean Dictionary, which will at least take care of the duplication. However, for ergonomics and discoverability reasons, I'd suggest moving the category back to `credit_cards`. 

/cc @gabrielluong @rocketsroger

**Comment by gabrielluong** [comment]
@wlach @rocketsroger We made the change from `credit_cards` into `metrics` when QA and our own testing did not find the `credit_card` in the metrics ping. See https://github.com/mozilla-mobile/fenix/issues/18711#issuecomment-850419521 for some context.

This is probably some implementation detail that is shared between Glean and Fenix, but we should try to make it work without grouping it all under `metrics`. It would be worthwhile to know if there were some `credit_cards` data that did it make it into Looker since that was the original implementation and was in Nightly for a short weekend.

**Comment by wlach** [comment]
> @wlach @rocketsroger We made the change from `credit_cards` into `metrics` when QA and our own testing did not find the `credit_card` in the metrics ping. See [#18711 (comment)](https://github.com/mozilla-mobile/fenix/issues/18711#issuecomment-850419521) for some context.
> 
> This is probably some implementation detail that is shared between Glean and Fenix, but we should try to make it work without grouping it all under `metrics`. It would be worthwhile to know if there were some `credit_cards` data that did it make it into Looker since that was the original implementation and was in Nightly for a short weekend.

I'm not sure what happened here, but some data definitely came through. You can now easily see how data appears in Looker or GLAM for many types of metrics, via the Glean Dictionary: 

![image](https://user-images.githubusercontent.com/20569/121773996-994fdb80-cb4d-11eb-9556-1c2418807f6b.png)
https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/credit_cards_autofill_card

It appears that some data did indeed show up for `credit_cards.autofill_card`:

https://mozilla.cloud.looker.com/explore/fenix/metrics?qid=x86w9AblneQ0II1Ytho8cr&toggle=fil,vis

> Our current worries is that if we don't group it under the `metrics` category then the metrics ping won't show up.

The category doesn't have anything to do with what pings a metric is sent in: it's just a bit of labelling designed to help with organization. If you have any questions about this, I'd definitely recommend asking on #glean on either Slack or Matrix. The Glean SDK team is pretty helpful. 😄

**Comment by rocketsroger** [comment]
Fixed with https://github.com/mozilla-mobile/fenix/pull/20909

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on the latest Nightly 94.0a1 09/21 using a Google Pixel 3a (Android 11)

✅  Tap add button through credit card management settings - Event Ping 629633a5-db7d-4620-bd38-ebc4580d71cd
```
        {
          "category": "credit_cards",
          "name": "management_add_tapped",
          "timestamp": 3763
        }
```

✅  Tap on a saved card through credit card management settings  - Event Ping 5f0810e1-feda-4872-a2d0-1874df3991b0
```
          "category": "credit_cards",
          "name": "management_card_tapped",
          "timestamp": 8290
        }
```
✅  Modified a credit card saved the changes using the "Save" button - Event Ping b5f3d883-c039-487a-a253-7a57b188ab08
```
        {
          "category": "credit_cards",
          "name": "modified",
          "timestamp": 19672
        }
```
✅  Modified a credit card saved the changes using the "✓" button - Event Ping 7c075fc4-6da1-45f2-a49b-ab218c44b968
```
    {
          "category": "credit_cards",
          "name": "modified",
          "timestamp": 34242
        }
```
✅  Navigate to a Credit card checkout website, tap the card number field, expand the prompt, confirm autofill - Event Ping 2d34bf6e-1597-4d5f-8568-ea3a0b6a7231
```
{
          "category": "credit_cards",
          "name": "form_detected",
          "timestamp": 10292
        },
        {
          "category": "credit_cards",
          "name": "autofill_prompt_shown",
          "timestamp": 10293
        },
        {
          "category": "credit_cards",
          "name": "autofill_prompt_expanded",
          "timestamp": 11733
        },
        {
          "category": "credit_cards",
          "name": "autofilled",
          "timestamp": 12674
        },
        {
          "category": "credit_cards",
          "name": "autofill_prompt_dismissed",
          "timestamp": 12675
        }
```

✅  Added 4 credit cards (2 using the "Save button", other 2 using the "✓" button ), deleted 2 (one using the trash bin icon, the other one using the "Delete card" button) - Metrics Ping 48697421-dabe-4f8a-99d4-6bc473b598ba 
```
"counter": {
          "credit_cards.deleted": 2,
          "credit_cards.saved": 4,
```

[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/AJ-090210)
[Logcat](https://github.com/mozilla-mobile/fenix/files/7202305/CC.txt)
